### PR TITLE
Professions and Scenario Update (CR) (Ready)

### PR DESCRIPTION
**Pull request by Noctifer-de-Mortem**
#### Summary
```SUMMARY: Balance "Changes various Professions and Scenarios to reflect recent changes."```

#### Purpose of change
Recent, relatively speaking, changes to how bionics work prompted me to make some slight changes to the professions that have them (example: changes to joint torsion ratchet bionic). From there I noticed other "inconsistencies", for lack of a better word, in professions that arose from other changes affecting play styles (example: butchering overhaul). This PR seeks to fix this as well as re-balance accordingly both professions and, to a much lesser extent, scenarios from both vanilla and the more professions and scenarios mod.

#### Describe the solution
Changes - Vanilla:

-Changed Bionic Prepper's `bio_torsionratchet` to `bio_furnace`:
This profession is mostly meant for inawoods where materials to keep themselves charge are plentiful but not "free" as `bio_torsionratchet` used to be since you still have to exert effort to gather and consume the materials.

-Changed Military Recruit's cost from 3 to 4, lowered it's rifle skill to 1 and gave it a dodging skill of 1:
 Cost raise due to profession already having rather good gear starting out including a gun with ammunition that is common, a melee weapon that can be attached to said gun to do reach attacks and that can be used on it's own effectively. Skill change based on the fact that training is meant to be basic (compared to other professions) combat performance since they are just a recruit but certainly dodging would be included.

-Changed Bionic Soldier's cost from 5 to 6 and made it's skill set match military recruit:
With skill set change, this profession is now like the military recruit but with better gear and bionics justifying cost change,

-Added `knife_hunting` to Bionic Sniper:
As another innawoods professions something to be able to butcher would be vital. The item itself is not good at combat and profession lacks skills to use it as such.

-Changes Bionic Assassin's `bio_torsionratchet` to `bio_metabolics` ~~and removed `dodge` skill~~:
As an assassin, this profession is meant to go somewhere prepared, take down whatever it needs to do and then leave; not meant for extended combat. In a pinch it's power source bionic was meant to give it emergency power to escape since without any power they are a sitting duck due to lack of skill. `bio_torsionratchet` is now a liability for a melee combat-centric profession like this one so `bio_metabolics`, which can still be used in a pinch and serves same role, makes sense to be used instead. ~~(See changes to CQB Bionic below for removal of dodge skill)~~

-Changed Commercial Cyborg's cost from 6 to 8, changed skills set to 4 electronics and 2 fabrication, and gave it light jacket to protect from the elements:
This profession seems like the urban version of the Bionic Prepper. Changed it's skills set so it can scavenge and mess with electronics right off the bat. Gave it the smartphone as it made sense given what it is and since it has the `bio_ups`. It has the tools and skills to craft things right of the bat and ability to not need a UPS justifies cost increase.'

-Changed Trapper's cost to 2 and gave it survival skill of 2:
A profession that says in description that spends lots of time outdoors; should have picked up a some skills from that as well as be able to butcher what they trap. They already have equipment to butcher, just needed the skill.

~~-Added `dodge` skill `bio_cqb`:
Makes sense for dodging to be part of the bionic since it is a combat skill. Deleted the 3 dodge skill from the bionic assassin since this will now be bundled with the bionic rendering a need for the skill gone.~~

Changes - More Classes and Scenarios Mod:

-Made the bionic special operator a combination of special operator and bionic soldier, lower cost to 8 and deleted `SCEN_ONLY` flag:
Gave it the skills set and gear of the operator only upgrading the smg in the same way it is upgraded from military recruit to bionic soldier, gave it the bionics of the bionic soldier and given the nerf from what it was before deleted the `SCEN_ONLY` flag and lower cost to 8 points.

-Lower Wasteland Cyborg's combat skills from 3 to 2:
This profession is more focused on what it can made rather than combat. Left it with enough skills to use it's equipment.

-Made Bionic Gangster a combination of Bionic Assassin and Gangster, lower cost from 10 to 8 and deleted `SCEN_ONLY` flag:
Gave it the gear and skills of the Gangster and Bionics of the Bionic Assassin. Took out `bio_cqb` and `bio_blade` and added `bio_targeting` since it fits the profession better. Given the nerf from what it was before lowered cost to 8 and deleted `SCEN_ONLY` flag.

-Added Wasteland Cyborg High Tech, Low Life scenario. 

#### Describe alternatives you've considered
What I would consider alternatives here would be telling myself to not do certain parts of the changes I did, no other ideas apart from not doing things arose. Any other ideas and suggestions will be considered as alternatives.

#### Additional context
Changes to `bio_cqb` unsuccessful so reverted.

### CODE REVIEW

**Comment by Noctifer-de-Mortem** [comment]
Game loads with no errors, and bot isn't telling me what's wrong...

**Comment by KorGgenT** [comment]
you need to run professions.json through the [linter](http://dev.narc.ro/cataclysm/format.html)

**Comment by Noctifer-de-Mortem** [comment]
Linted the file, game runs, lets see if that fixes it.

File: data/json/professions.json [hunk 1]
```diff
@@ -1526,12 +1528,12 @@
     "type": "profession",
     "ident": "bionic_customer",
     "name": "Commercial Cyborg",
-    "description": "You always had to have the latest and best gadgets and gizmos, so is it any wonder that you upgraded your flesh along with your smart phone?  Only time will tell if your status as a marvel of bionic technology will be enough to ensure your survival after the apocalypse.",
-    "points": 6,
+    "description": "You always had to have the latest and best gadgets and gizmos, so is it any wonder that you upgraded your flesh along with your smart phone?  Only time will tell if your passion for electronics and your status as a marvel of bionic technology will be enough to ensure your survival after the apocalypse.",
+    "points": 8,
     "CBMs": [ "bio_flashlight", "bio_tools", "bio_ups", "bio_watch", "bio_batteries", "bio_power_storage_mkII" ],
-    "skills": [ { "level": 1, "name": "electronics" } ],
+    "skills": [ { "level": 4, "name": "electronics" }, { "level": 2, "name": "fabrication" } ],
```
Comment (I-am-Erk) [inline data/json/professions.json]: I'm don't it makes a lot of sense for a consumer end cyborg to be an 8 point build with high starting skills. This is the equivalent to the person who always has the latest iPhone, not a tech guru.

File: data/mods/more_classes_scenarios/cs_classes.json [hunk 2]
```diff
@@ -322,43 +318,37 @@
...
-      "bio_blade",
       "bio_shock",
+      "bio_carbon",
+      "bio_adrenaline",
       "bio_night_vision",
-      "bio_torsionratchet",
+      "bio_face_mask",
+      "bio_metabolics",
       "bio_power_storage_mkII"
     ],
```
Comment (I-am-Erk) [inline data/mods/more_classes_scenarios/cs_classes.json]: I would say the gangster's previous set of bionics was quite flavourful. What's the rationale in replacing it with an assassin's set of bionics? This is someone geared towards intimidation and thuggery, finger blades and sunglasses make sense.

File: data/json/professions.json (hunk 1 above)
Comment (Noctifer-de-Mortem) [reply data/json/professions.json]: Reasoning is that they arr really into the latest tech so they would know about it. For example, how to install a ups mod to their electronics to be able to use it from their bionic's power source.

File: data/mods/more_classes_scenarios/cs_classes.json (hunk 2 above)
Comment (Noctifer-de-Mortem) [reply data/mods/more_classes_scenarios/cs_classes.json]: It was always meant to be a fusion of gangster and bionic assassin but when creating it I added way more than that and justified it with a higher point cost and locking it to a specific scenario.

**Pull request merged**